vibi-dpu/src/db/prs.rs

```diff
@@ -104,4 +104,62 @@ pub async fn get_pr_info_from_db(workspace_slug: &str, repo_slug: &str, pr_numbe
     return Some(pr_info);
-}
\ No newline at end of file
+}
+
+pub async fn github_process_and_update_pr_if_different(webhook_data: &Value, repo_owner: &str, repo_name: &str, pr_number: &str, repo_provider: &str) -> bool {
+    println!("[github_process_and_update_pr_if_different] {:?}, {:?}, {:?}, {:?}", repo_owner, repo_name, pr_number, repo_provider);
+    let event_action = webhook_data["action"].to_string().trim_matches('"').to_string();
+
+    let pr_info_parsed_opt = parse_github_webhook_data(webhook_data);
+    if pr_info_parsed_opt.is_none() {
+        eprintln!("[github_process_and_update_pr_if_different] Unable to parse webhook data");
+        return false;
+    }
+    let pr_info_parsed = pr_info_parsed_opt.expect("Empty pr_info_parsed_opt");
+
+    if event_action == "opened" {
+        // new PR opened
+        println!("[github_process_and_update_pr_if_different|new_pr_opened] {:?}", pr_info_parsed);
+        update_pr_info_in_db(&repo_owner, &repo_owner, &pr_info_parsed, &pr_number, repo_provider).await;
+        return true;
+    }
+    if event_action == "synchronize" {
+        println!("[github_process_and_update_pr_if_different| event_action synchronise] pr_info_parsed: {:?}", &pr_info_parsed);
+        update_pr_info_in_db(&repo_owner, &repo_name, &pr_info_parsed, &pr_number, repo_provider).await;
+        return true // commits are different, and PR info should be updated
+    } 
+    if event_action == "submitted" {
+        let event_review_status = webhook_data["review"]["state"].to_string().trim_matches('"').to_string();
+        if event_review_status == "approved" {
+            println!("[github_process_and_update_pr_if_different| pr has been approved] pr_info_parsed: {:?}", &pr_info_parsed);
+            update_pr_info_in_db(&repo_owner, &repo_name, &pr_info_parsed, &pr_number, repo_provider).await;
+            return true
+        } else {
+            println!("[github_process_and_update_pr_if_different|no_update_needed] event is not approved");
+            return false; // event is not open or synchronize
+        }
+    } else {
+        println!("[github_process_and_update_pr_if_different | no update needed] event is not opened or synchronise or approved");
+        return false;
+    }
+
+}
+
+fn parse_github_webhook_data(webhook_data: &Value) -> Option<PrInfo> {
+    println!("[parse_github_webhook_data] webhook_data: {:?}", &webhook_data);
+    let pr_head_commit_raw = webhook_data["pull_request"]["head"]["sha"].to_string();
+    let pr_head_commit = pr_head_commit_raw.trim_matches('"');
+    let base_head_commit_raw = webhook_data["pull_request"]["base"]["sha"].to_string();
+    let base_head_commit = base_head_commit_raw.trim_matches('"');
+    let pr_state_raw = webhook_data["pull_request"]["state"].to_string();
+    let pr_state = pr_state_raw.trim_matches('"');
+    let pr_branch_raw = webhook_data["pull_request"]["head"]["ref"].to_string();
+    let pr_branch = pr_branch_raw.trim_matches('"');
```

Comment:
Creating the `raw` versions of these variables seems unnecessary. Why not directly do `.to_string().trim_matches('"').to_string()`?

This is a pattern that we have followed in many other places. And you wouldn't have to do the `to_string()` in lines 159-161 as well.